File: samples/Test/Classes/Control/PageViewTest.cpp

#include "PageViewTest.h"

PageViewTest::PageViewTest()
{
    this->setTitle("CAImageView");
}

PageViewTest::~PageViewTest()
{
}

void PageViewTest::viewDidLoad()
{
    
    std::vector<CAView*> vec =
    {
        CAImageView::createWithImage(CAImage::create("image/1.jpg")),
        CAImageView::createWithImage(CAImage::create("image/2.jpg")),
        CAImageView::createWithImage(CAImage::create("image/3.jpg"))
    };

    m_pPageView = CAPageView::createWithLayout(DLayoutFill, CAPageView::Orientation::Horizontal);
    m_pPageView->setViews(CAVector<CAView*>(vec));
    this->getView()->addSubview(m_pPageView);
    
    m_pPageView->onEndTurning(CALLBACK_BIND_0(PageViewTest::pageViewDidEndTurning, this));
    
    m_pPageControl = CAPageControl::createWithLayout(DLayout(DHorizontalLayout_W_C(100, 0.5), DVerticalLayout_T_H(30, 50)));
    m_pPageControl->setNumberOfPages((int)vec.size());
    m_pPageControl->setPageIndicatorTintColor(CAColor4B::GRAY);
    m_pPageControl->setCurrentPageIndicatorTintColor(CAColor4B::GREEN);
    m_pPageControl->setTarget([&](int index)
    {
        m_pPageView->setCurrPage(index, true);
    });
    this->getView()->addSubview(m_pPageControl);
}

void PageViewTest::viewDidUnload()
{
    // Release any retained subviews of the main view.
    // e.g. self.myOutlet = nil;
}



void PageViewTest::pageViewDidEndTurning()
{
    m_pPageControl->setCurrentPage(m_pPageView->getCurrPage());
    m_pPageControl->updateCurrentPageDisplay();
}

File: samples/Test_JS/Resources/js/Control/PageViewTest.js
/**
 * Created by crossApp on 16/8/3.
 */
var PageViewTest = ca.CAViewController.extend({

    ctor: function () {
        this._super();
    },

    viewDidLoad: function() {

        var view1 = ca.CAImageView.createWithImage(ca.CAImage.create("image/1.jpg"));
        var view2 = ca.CAImageView.createWithImage(ca.CAImage.create("image/2.jpg"));
        var view3 = ca.CAImageView.createWithImage(ca.CAImage.create("image/3.jpg"));
        var views = new Array();
        views.push(view1);
        views.push(view2);
        views.push(view3);

        this._pageView = ca.CAPageView.createWithLayout(ca.DLayout.fill(), 0);
        this._pageView.setViews(views);
        this.getView().addSubview(this._pageView);

        this._pageView.onBeginTurning(this.pageViewDidBeginTurning.bind(this)) ;
        this._pageView.onEndTurning(this.pageViewDidEndTurning.bind(this)) ;
        this._pageView.onDidSelectedPageAtIndex(this.pageViewDidSelectPageAtIndex.bind(this)) ;


        this.pageControl = ca.CAPageControl.createWithLayout(ca.DLayout.set(ca.DHorizontalLayout_W_C(100, 0.5), ca.DVerticalLayout_T_H(30, 50)));
        this.pageControl.setNumberOfPages(views.length);
        this.pageControl.setPageIndicatorTintColor(ca.CAColor4B.GRAY);
        this.pageControl.setCurrentPageIndicatorTintColor(ca.CAColor4B.GREEN);
        this.getView().addSubview(this.pageControl);

        var pageContro = this.pageControl;
        var pageView = this._pageView;
        this.pageControl.setTarget(function(index)
        {
            ca.log("Button RoundedRect");
            ca.log("btn_tag==="+pageContro.getCurrentPage());
            pageView.setCurrPage(pageContro.getCurrentPage(), true);
        });
    },

    pageViewDidBeginTurning: function () {

        ca.log("pageViewDidBeginTurning()" );

    },
    pageViewDidEndTurning: function () {
        this.pageControl.setCurrentPage(this._pageView.getCurrPage());
        this.pageControl.updateCurrentPageDisplay();
        ca.log("pageViewDidEndTurning() ---  this._pageView.getCurrPage() = " + this._pageView.getCurrPage() );
    },
    pageViewDidSelectPageAtIndex: function (index, point) {
        ca.log("pageViewDidSelectPageAtIndex("+index+"),("+point+")");
    }


});
